app/app_iface.go

```diff
@@ -41,12 +41,22 @@ type AppIface interface {
 	ListAutocompleteCommands(teamID string, T i18n.TranslateFunc) ([]*model.Command, *model.AppError)
 	// @openTracingParams teamID, skipSlackParsing
 	CreateCommandPost(post *model.Post, teamID string, response *model.CommandResponse, skipSlackParsing bool) (*model.Post, *model.AppError)
+	// AddChannelMember adds a user to a channel. It is a wrapper over AddUserToChannel.
+	// checkTeamMember is used to indicate whether it will check for the user was deleted from
+	// the team or not. This is useful to avoid in scenarios when we just added the team member,
+	// and thereby know that there is no need to check this.
+	AddChannelMember(userID string, channel *model.Channel, opts ChannelMemberOpts) (*model.ChannelMember, *model.AppError)
 	// AddCursorIdsForPostList adds NextPostId and PrevPostId as cursor to the PostList.
 	// The conditional blocks ensure that it sets those cursor IDs immediately as afterPost, beforePost or empty,
 	// and only query to database whenever necessary.
 	AddCursorIdsForPostList(originalList *model.PostList, afterPost, beforePost string, since int64, page, perPage int, collapsedThreads bool)
 	// AddPublicKey will add plugin public key to the config. Overwrites the previous file
 	AddPublicKey(name string, key io.Reader) *model.AppError
+	// AddUserToChannel adds a user to a given channel.
+	// checkTeamMember is used to indicate whether it should be checked that a user has
+	// already been removed from that team or not. This is useful to avoid in scenarios
+	// when we just added the team member, and thereby know that there is no need to check this.
+	AddUserToChannel(user *model.User, channel *model.Channel, doNotCheckTeamMember bool) (*model.ChannelMember, *model.AppError)
```

Comment:
What do you think about something like this to avoid the double-negative? In the `false` case it'll read like "don't skip team member integrity check" vs. "don't do not check team member":

```suggestion
	AddUserToChannel(user *model.User, channel *model.Channel, skipTeamMemberIntegrityCheck bool) (*model.ChannelMember, *model.AppError)
```

Reply:
Sounds good to me.